crates/gpio/src/backend.rs

```diff
@@ -197,7 +227,13 @@ mod tests {
 
     impl DeviceConfig {
         pub fn new_with(devices: Vec<u32>) -> Self {
-            DeviceConfig { inner: devices }
+            let gpio_devices: Vec<GpioDeviceConfig> = devices
+                .into_iter()
+                .map(GpioDeviceConfig::PhysicalDevice)
+                .collect();
+            DeviceConfig {
+                inner: gpio_devices,
+            }
```

Comment:
What about this instead ?

             DeviceConfig {
                 inner: devices
                     .into_iter()
                     .map(GpioDeviceConfig::PhysicalDevice)
                     .collect(),
             }

Reply:
ok